coreTable/OwidTable.ts

```diff
@@ -562,6 +564,62 @@ export class OwidTable extends CoreTable<OwidRow, OwidColumnDef> {
         )
     }
 
+    interpolateColumnsLinearly(columnSlug: ColumnSlug) {
```

Comment:
I don't know quite how, but it seems these interpolation functions could be broken up a bit into chains of smaller functions, and maybe share more code?

Any ideas @danielgavrilov ?

Reply:
Yeah it feels very similar, but there a few differences that might make the generalization tangled. E.g. the tolerance one distinguishes timeColumnOfTable and timeColumnOfValue, in order to be idempotent.

I'm not sure by looking at it – if Marcel wants to give it a go, great, but otherwise I think we can move onto other issues.

Reply:
Another thing is, `complete()` can inject a lot of unnecessary rows for tolerance implementation, it's the slowest part of the transform. We might want to optimize that and change the implementation.

Reply:
I tried my best, but as Daniel said it's really difficult.
One interpolation method alters one column, the other does two; one inserts a new column, the other doesn't; one differentiates between `timeColumnOfTable` and `timeColumnOfValue`, the other doesn't care -- all of these issues make it increasingly difficult to generalize here.
I still think a generalization and some thinking there would be very valuable because we really do have quite some duplicate code here. But let's move that until after the merge.

Reply:
Sounds good!